hw/chisel/doc/xdma.md

```diff
@@ -0,0 +1,238 @@
+|     960     |     RW    |      The LSB of Source Base Address      |              [31:0]              |
+|     961     |     RW    |      The MSB of Source Base Address      |              [15:0]              |
+|     962     |     RW    |       Spatial Bound at Source side       |     0, 1, 2, 3, 4, 5, 6, 7, 8    |
+|     963     |     RW    |    Temporal Bound (D1) at Source side    |              [15:0]              |
+|     964     |     RW    |    Temporal Bound (D2) at Source side    |              [15:0]              |
+|     965     |     RW    |       Spatial Stride at Source side      |              [31:0]              |
+|     966     |     RW    |    Temporal Stride (D1) at Source side   |              [31:0]              |
+|     967     |     RW    |    Temporal Stride (D2) at Source side   |              [31:0]              |
+|     968     |     RW    |    The LSB of Destination Base Address   |              [31:0]              |
+|     969     |     RW    |    The MSB of Destination Base Address   |              [15:0]              |
+|     970     |     RW    |     Spatial Bound at Destination side    |     0, 1, 2, 3, 4, 5, 6, 7, 8    |
+|     971     |     RW    |  Temporal Bound (D1) at Destination side |              [15:0]              |
+|     972     |     RW    |  Temporal Bound (D2) at Destination side |              [15:0]              |
+|     973     |     RW    |    Spatial Stride at Destination side    |              [31:0]              |
+|     974     |     RW    | Temporal Stride (D1) at Destination side |              [31:0]              |
+|     975     |     RW    | Temporal Stride (D2) at Destination side |              [31:0]              |
+|     976     |     RW    |  Byte Mask (Strobe) at Destination side  |               [7:0]              |
+|     977     |     RW    |    Bypass Enabler of  Memset Extension   |               0, 1               |
+|     978     |     RW    |        Memset Extension Custom CSR       |               [7:0]              |
+|     979     |     RW    |    Bypass Enabler of MaxPool Extension   |               0, 1               |
+|     980     |     RW    |       MaxPool Extension Custom CSR       |               [7:0]              |
+|     981     |     RW    |   Bypass Enabler of Transpose Extension  |               0, 1               |
+|     982     |     RW    |       Commit Current Configuration       |                 1                |
+|     983     |     R     |      Committed Configuration Counter     |              [31:0]              |
+|     984     |     R     |      Finished Configuration Counter      |              [31:0]              |
+
+Offsets may vary based on the configuration, but the Chisel code will automatically generate the necessary C headers, ensuring ease of use for developers.
+
+## Design of Custom Datapath Extension
+
+### Native Datapath Extension
+
+Designing a Datapath Extension involves four steps:
+
```

Comment:
Do the extensions have a handshake with the input data or should it be a 0-cycle delay datapath?

Reply:
There is handshaking. You can design it in full-combinational style, but you can also add arbitrary latency to it, as there is handshake signal. 

MaxPool extension is with the latency. Furthermore, the input length and output length are not equal. XDMA also supports it. 

Transposer extension is with no latency. However, XDMA will still add one delay implicitly, to prevent critical path being too long after the concatenation of many combinational extension.